RecipeApp.Core/RecipeConverter.cs

```diff
@@ -1,14 +1,176 @@
+        static List<string> doubleUnits = new List<string>();
+
+        // Entry method to get some initial setup for Lists
+        public static void InitialSetup() {
+            PopulateDoubleUnitsList();
+            SortCookingUnitsByMultiplier();
+        }
+
+        public static string RecipeValidation(string recipeText) {
+            if (recipeText.Length == 0)
+            {
+                return "Empty recipes are not allowed.";
+            }
+            else if (!recipeText.Any(char.IsDigit))
+            {
+                return "Recipe does not contain any numeric values.";
+            }
+            else if (!RecipeConverter.DoesContainCookingUnits(recipeText))
+            {
+                return "Recipe does not contain any cooking units";
+            }
+            else if (RecipeConverter.ReturnFirstFaultyCookingUnitIndex(recipeText) != -1)
+            {
+                return "Bad or non-existant cooking unit values detected.";
+            }
+            return "";
+        }
+
+        // Sort cooking units by multiplier
+        static void SortCookingUnitsByMultiplier() {
+            bool sorted = true;
+            for (int j = 0; j < multipliers.Length; j++)
+            {
+                for (int i = 0; i < multipliers.Length - 1; i++)
+                {
+                    if (multipliers[i]>multipliers[i+1])
+                    {
+                        //swapping multipliers
+                        double temp = multipliers[i];
+                        multipliers[i] = multipliers[i + 1];
+                        multipliers[i + 1] = temp;
+                        //swapping units
+                        string strTemp = units[i];
+                        units[i] = units[i + 1];
+                        units[i + 1] = strTemp;
+
+                        sorted = false;
+                    }
+                }
+                if (sorted)
+                {
+                    return;
+                }
+            }
+        }
+
+        // Validation method to check if string contains any of the cooking units
+        public static bool DoesContainCookingUnits(string textToCheck) {
+            bool containsUnits = false;
+            List<string> wordList = new List<string>(textToCheck.ToLower().Split(' ', '\n'));
+            for (int i = 0; i < units.Length; i++)
+            {
+                if (wordList.Contains(units[i]))
+                {
+                    containsUnits = true;
+                }
+            }
+            if (wordList.Contains("ml") || wordList.Contains("l"))
+            {
+                containsUnits = true;
+            }
+            return containsUnits;
+        }
+
+        // Check method to ensure that each unit as some value number before it
+        // returns -1 if ok, else returns index of bad cooking unit
+        public static int ReturnFirstFaultyCookingUnitIndex(string textToCheck) {
```

Comment:
Avoid using a word `Return` in a method name. It sounds forced. Try to be more natural, for example:
`FindFirstFaultyCookingUnitIndex`